lib/trento/application/process_managers/deregistration_process_manager.ex

```diff
@@ -84,10 +144,47 @@ defmodule Trento.DeregistrationProcessManager do
     ]
   end
 
-  def apply(%DeregistrationProcessManager{} = state, %ClusterRolledUp{
-        cluster_id: cluster_id
-      }) do
-    %DeregistrationProcessManager{state | cluster_id: cluster_id}
+  def handle(
+        %DeregistrationProcessManager{
+          cluster_id: cluster_id,
+          database_instances: database_instances,
+          application_instances: application_instances
+        },
+        %HostDeregistrationRequested{
+          host_id: host_id,
+          requested_at: requested_at
+        }
+      ) do
+    List.flatten([
+      %DeregisterClusterHost{
```

Comment:
We still have the scenario where this `cluster_id` could be nil. We need to take care of that.